provider/pkg/provider/environment.go

```diff
@@ -144,14 +158,27 @@ func (st *PulumiServiceEnvironmentResource) Create(req *pulumirpc.CreateRequest)
 	}
 
 	// First check if yaml is valid
-	_, diagnostics, err := st.client.CheckYAMLEnvironment(context.Background(), input.OrgName, []byte(input.Yaml))
+	_, diagnostics, err := st.client.CheckYAMLEnvironment(context.Background(), input.OrgName, []byte(input.Yaml), esc_client.CheckYAMLOption{})
+	if diagnostics != nil {
+		return nil, fmt.Errorf("failed to check environment, yaml code failed following checks: %+v", diagnostics)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to check environment due to error!: %+v", err)
+	}
```

Comment:
Did I really miss err checks on this one o_O
Ty for adding them!

Reply:
actually this looks like the result of a rebase. `diagnostics` is no longer being used here so i can remove that check (but could keep the err check still)